plugins/net_plugin/net_plugin.cpp

```diff
@@ -2565,13 +2613,9 @@ namespace eosio {
             close( false );
             return;
          }
-
-         boost::asio::async_read( *socket,
-            pending_message_buffer.get_buffer_sequence_for_boost_async_read(), completion_handler,
-            boost::asio::bind_executor( strand,
-              [conn = shared_from_this(), socket=socket]( boost::system::error_code ec, std::size_t bytes_transferred ) {
+         auto handle_read = [conn = shared_from_this(), psocket=&get_socket()]( boost::system::error_code ec, std::size_t bytes_transferred ) {
```

Comment:
This needs to capture the shared_ptr of the socket so it maintains the lifetime of the socket.